### PR TITLE
Self Review - Role  - Admin view -UI

### PR DESCRIPTION
**Pull request by surekhag**

### CODE REVIEW

File: client/src/components/selfReviewDetails/SelfReviewDetails.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      selfReviewDeatails.due_to ? selfReviewDeatails.due_to.slice(0, 10) : null
+    ],
+    ['Feedback', selfReviewDeatails.feedback],
+    ['Status', selfReviewDeatails.status],
+    ['Review Form Link', selfReviewDeatails.review_form_link]
+  )
+
+  const [selectedStatus, setSelectedStatus] = useState()
+  const userSelfReviewUpdateError = useSelector(
+    state => state.selfReviewReducer.userSelfReviewUpdateError
```
Comment (bhushan-oe) [inline client/src/components/selfReviewDetails/SelfReviewDetails.js]: Selector functions?

File: client/src/components/selfReviewDetails/SelfReviewDetails.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    [
+      'Due To',
+      selfReviewDeatails.due_to ? selfReviewDeatails.due_to.slice(0, 10) : null
+    ],
+    ['Feedback', selfReviewDeatails.feedback],
+    ['Status', selfReviewDeatails.status],
+    ['Review Form Link', selfReviewDeatails.review_form_link]
+  )
+
+  const [selectedStatus, setSelectedStatus] = useState()
```
Comment (bhushan-oe) [inline client/src/components/selfReviewDetails/SelfReviewDetails.js]: set default value for state

File: client/src/sagas/watchers/peerReviewsForUser.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  setAllPeerForUser,
+  setAllPeerForUserError
+} from '../../actions/peerReviewAction'
+
+function* workerLoadUserPeerReviewSaga() {
+  try {
+    const peerReviews = yield call(loadAllUserPeerReviews)
+    yield put(setAllPeerForUser(peerReviews))
+  } catch (e) {
+    console.log(e)
```
Comment (bhushan-oe) [inline client/src/sagas/watchers/peerReviewsForUser.js:14]: console log required?

File: client/src/sagas/watchers/updatePeerReview.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    yield put(setUpdateReviewError(e))
+  }
+}
+
+export function* watchUpdatePeerReviewSaga() {
+  yield takeLatest(UPDATE_PEER_REVIEW, workerUpdatePeerReviewSaga)
+}
+function* workerDaletePeerReviewSaga(data) {
+  try {
+    console.log(data, 'datain SAga')
```
Comment (bhushan-oe) [inline client/src/sagas/watchers/updatePeerReview.js:32]: console log

File: client/src/views/Dashboard/Dashboard.js [hunk 5]
```diff
@@ -17,16 +18,22 @@ import Card from '../../components/Card/Card.js'
...
+import { loadAllEmployeeData } from '../../actions/employeeAction.js'
 
 const useStyles = makeStyles(styles)
 const Dashboard = props => {
   const classes = useStyles()
   const [searchText, setsearchText] = useState('')
   const [showDetail, setShowDetail] = useState(false)
+  const [showSelfReviewDetail, setShowSelfreviewDetail] = useState(false)
   const [peerDetails, setPeerDetails] = useState('')
+  const [selfReviewDetails, setSelfReviewDetails] = useState()
```
Comment (bhushan-oe) [inline client/src/views/Dashboard/Dashboard.js]: set default value for state

File: client/src/views/Dashboard/Dashboard.js [hunk 6]
```diff
@@ -35,14 +42,35 @@ const Dashboard = props => {
...
+
+  const SelfReviewListingHeader = [
+    'Projects',
+    'From date',
+    'To date',
+    'Due Fom Date',
+    'Status'
+  ]
   const peerReviews = useSelector(
     state => state.peerReviewReducer.userPeerReview
```
Comment (bhushan-oe) [inline client/src/views/Dashboard/Dashboard.js]: selector functions as mentioned on chat

File: client/src/views/Dashboard/Dashboard.js [hunk 7]
```diff
@@ -64,23 +92,65 @@ const Dashboard = props => {
...
+  let projectsArr = []
+  if (
+    userSelfReviews &&
+    userSelfReviews.length > 0 &&
+    userReviewDetailsArr.length == 0 &&
+    projectData
+  ) {
+    userSelfReviews.map((review, key1) => {
+      projects = review.project_ids.split(',')
+      projectData.map((item, key) => {
```
Comment (bhushan-oe) [inline client/src/views/Dashboard/Dashboard.js:105]: Loop inside loop, is it required? can we reduce this? There are two hooks for cache. useCallback and useMemo. Can you use that?

File: server/app/api/controllers/self_review.js [hunk 8]
```diff
@@ -0,0 +1,125 @@
+const Self_Review_Model = require("../models/self_review");
+module.exports = {
+  create: function(req, res, next) {
+    Self_Review_Model.create(
+      {
+        employee_id: req.body.employee_id,
```
Comment (bhushan-oe) [inline server/app/api/controllers/self_review.js]: spread operator

**Review by bhushan-oe — CHANGES_REQUESTED** [review]

File: client/src/components/selfReviewDetails/SelfReviewDetails.js [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+const useStyles = makeStyles(styles)
+
+const SelfReviewDetails = props => {
+  const classes = useStyles()
+  const { selfReviewDeatails, projectDetails, ClickHandler } = props
+  const { addToast } = useToasts()
+  const dispatch = useDispatch()
+  const selfReviewDetailHeader = ['Self Review Details', '']
+  let temp = []
+  temp.push(
```
Comment (bhushan-oe) [inline client/src/components/selfReviewDetails/SelfReviewDetails.js]: what use of temp?

File: client/src/components/selfReviewDetails/SelfReviewDetails.js (hunk 2 above)
Comment (surekhag) [reply client/src/components/selfReviewDetails/SelfReviewDetails.js]: Done

File: client/src/sagas/watchers/peerReviewsForUser.js (hunk 3 above)
Comment (surekhag) [reply client/src/sagas/watchers/peerReviewsForUser.js:14]: @RakshandaWankhede  has fixed this in separate branch. ..

File: client/src/components/selfReviewDetails/SelfReviewDetails.js (hunk 1 above)
Comment (surekhag) [reply client/src/components/selfReviewDetails/SelfReviewDetails.js]: Done

**Review by bhushan-oe — CHANGES_REQUESTED** [review]

File: client/src/views/Dashboard/Dashboard.js (hunk 5 above)
Comment (surekhag) [reply client/src/views/Dashboard/Dashboard.js]: Done

Comment (surekhag) [reply client/src/views/Dashboard/Dashboard.js]: Done

File: client/src/views/Dashboard/Dashboard.js (hunk 6 above)
Comment (surekhag) [reply client/src/views/Dashboard/Dashboard.js]: @RakshandaWankhede  will update in PR #9

Comment (surekhag) [reply client/src/views/Dashboard/Dashboard.js]: Done

File: client/src/components/selfReviewDetails/SelfReviewDetails.js (hunk 9 above)
Comment (surekhag) [reply client/src/components/selfReviewDetails/SelfReviewDetails.js]: Renamed .

File: client/src/views/Dashboard/Dashboard.js (hunk 7 above)
Comment (surekhag) [reply client/src/views/Dashboard/Dashboard.js:105]: We are going to update get api so this code we will remove afterwords.

Comment (surekhag) [reply client/src/views/Dashboard/Dashboard.js:105]: Not fixing it .

File: server/app/api/controllers/self_review.js (hunk 8 above)
Comment (surekhag) [reply server/app/api/controllers/self_review.js]: Destructuring ? you mean . This is done.

File: client/src/sagas/watchers/updatePeerReview.js (hunk 4 above)
Comment (RakshandaWankhede) [reply client/src/sagas/watchers/updatePeerReview.js:32]: Done . resolved in another branch

**Comment by surekhag** [comment]
Already merged to admin-view branch hence closing this PR.

**Pull request closed without merging**